contracts/contracts/GrantRegistry.sol

```diff
@@ -0,0 +1,135 @@
+pragma solidity ^0.8.5;
+
+contract GrantRegistry {
+  // Grant ID - The hash of the grants meetadata
+  bytes32 public grantRegistryId;
+  // Metadata Pointer
+  string public metaPtr;
+
+  // Owner
+  address owner;
+
+  modifier onlyOwner() {
+    require(msg.sender == owner);
+    _;
+  }
+
+  // TODO: Only allow the owner (which is the curator) address to configure the contract
+  // while its in config mode, then switch to active. Once active, the owner can only
+  // close the registry (have a conversation around this approach)
+  constructor(
+    address _owner,
+    bytes32 _grantRegistryId,
+    string memory _metaPtr
+  ) {
+    owner = _owner;
+    //state = State.CONFIG;
+    grantRegistryId = _grantRegistryId;
+    metaPtr = _metaPtr;
+  }
+
+  // State
+  enum State {
+    CONFIG,
+    ACTIVE,
+    CLOSED
+  }
+  //State public state;
+
+  // Grant Pointer Object
+  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
+  // TODO: Implement safemath for avoiding overflows with ints
+  uint256 public grantCount = 0;
+  mapping(bytes32 => Grant) public grants;
+  bytes32[] public grantIds;
+
+  struct Grant {
```

Comment:
Should the data about a Grant live in a Registry, or should it live in some external structure, where a Registry simply collects a list of Grant Unique identifiers? This question relates to the general question above about what the purpose of a GrantRegistry is.

An alternative to this approach would be for each Grant to be a contract instance that conforms to some `GrantInterface` requirement. This solves the unique identifier problem: the contract instance becomes the identifier. It would also allow for more flexibility and extensibility of the functionality around a Grant itself (imagine for example a Grant instance that is itself a mini DAO whose members have control of the parameters which define that grant and where the funds go). Furthermore, if we use something like UpgradeableProxies, it might provide a migration path for Grant owners to migrate their Grant contract over the course of our iteration with this product.

Another alternative (or perhaps orthogonal) approach would be to make each Grant instance an ERC721 NFT that could be transferred. In this instance, the ERC721 contract could be a sort of "global registry" of all Grants, where Rounds/Registry could collect the ERC721 identifiers.

Reply:
Another thought if grants are deployed from a factory, the unique ID can just be counter equal to `grantCount`. The first time a new grant is deployed it has an ID of 0 since `grantCount = 0`, next one has ID of 1

Reply:
I think my comment above covers most of this.

Regarding the ID, the count makes sense to me.